src/main/java/com/github/algobot76/surabaya/util/StaticAnalysisVisitor.java

```diff
@@ -49,7 +52,7 @@ public void visit(CompilationUnit n, Project project) {
 			packageName = ((PackageDeclaration) packageDeclaration.get()).getName().toString();
 		}
 		Package newPackage = new Package();
-		project.addPackage(packageName, newPackage);
+		newPackage = project.addPackage(packageName, newPackage);
```

Comment:
There is no need to return the package from `addPackage` because only the reference to the `newPackage` is stored in the map inside the `project`.

Reply:
Oh, we need a return in case the package for that name already exists. But yeah, it was pretty poorly implemented. I've refactored it now to make it clearer